service/history/replication/executable_activity_state_task.go

```diff
@@ -133,6 +133,7 @@ func (e *ExecutableActivityStateTask) HandleErr(err error) error {
 	case *serviceerrors.RetryReplication:
 		namespaceName, _, nsError := e.GetNamespaceInfo(e.NamespaceID)
 		if nsError != nil {
+			// todo: nsError is lost here
 			return err
```

Comment:
Oof. Another reason why there should only be one error ever in scope. If there's two, we should just extract a function.

Reply:
cc @yux0 
I think those operations are just best effort when handling the original replication task processing error and we always need to return the original processing error?

Reply:
Yes. It just return the original processing error but we should log those intermediate errors.